src/Lti/Extractor/ReplaceResultSourceIdExtractor.php

```diff
@@ -39,12 +40,12 @@ public function __construct(string $xmlNamespace)
     /**
      * @throws InvalidLtiReplaceResultBodyException
      */
-    public function extractSourceId(string $xmlContent): int
+    public function extractSourceId(string $xmlContent): UuidV6
     {
         try {
             $xml = new SimpleXMLElement($xmlContent);
         } catch (Throwable $exception) {
-            throw new InvalidLtiReplaceResultBodyException();
+            throw new InvalidLtiReplaceResultBodyException('Invalid XML received.');
```

Comment:
Can you also specify the previous exception. EX: 
```
throw new InvalidLtiReplaceResultBodyException('Invalid XML received.', 0, $exception);

Reply:
Fixed